### DISCUSSION TITLE
Quarkus ConfigItems - how to use no prefix?

### DISCUSSION THREAD

**Discussion by rsshah** [quarkusio/quarkus#31881] [Q&A] [answered]
I have a requirement that I need an extension's config property to be named exactly `myProperty` in my application config. When I use a `@ConfigRoot` and `@ConfigItem` class as such:

```
@ConfigRoot(prefix="", phase=ConfigPhase.RUN_TIME, name="")
public class MyConfig {
    /**
     * javadoc
     */
    @ConfigItem(name= "myProperty")
    String myProperty;
}
```

Quarkus prefixes a `.` and looks for `.myProperty` in configuration instead, throwing this error:

```
One or more configuration errors have prevented the application from starting. The errors are:
  - SRCFG00014: The config property .myProperty is required but it could not be found in any config source

```

How do I just tell it what property name I want it to use, without it trying to be too smart?

I've found myself just ending up referencing the `ConfigProvider` in my config objects to work around this, which seems like an anti-pattern, but I don't know the idiomatic way to do what I want in Quarkus. I'm still learning the framework.

```
@ConfigRoot(prefix="", phase=ConfigPhase.RUN_TIME, name="")
public class MyConfig {
    public String getMyProperty() {
        return ConfigProvider.getConfig().getValue("myProperty", String.class);
    }

// other potential configs...
}
```

**Comment by radcortez** [comment, marked as answer]
Please, use `@ConfigMapping` and interface-based configs: https://quarkus.io/guides/writing-extensions#configuration.

You can use `@WithName` to override the configuration name, and it should work with empty prefixes.
